driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -81,4 +100,27 @@ public static ArrayExpression<BooleanExpression> ofBooleanArray(final boolean...
         return new MqlExpression<>((cr) -> new BsonArray(result));
     }
 
+
+    public static ArrayExpression<IntegerExpression> ofIntegerArray(final int... ofIntegerArray) {
+        List<BsonValue> array = Arrays.stream(ofIntegerArray)
+                .mapToObj(BsonInt32::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new BsonArray(array));
+    }
+
+    public static DocumentExpression ofDocument(final Bson document) {
+        if (document == null) {
+            throw new IllegalArgumentException("document cannot be null");
+        }
+        // All documents are wrapped in a $literal. It is possible to check
+        // for empty docs (see https://jira.mongodb.org/browse/SERVER-46422)
+        // and to traverse for $$, but doing so would be unsafe and brittle.
+        return new MqlExpression<>((cr) -> document == null
```

Comment:
Why are we checking for null here if we throw on null just above?

Reply:
Looks like I fixed this in an ensuing PR: https://github.com/mongodb/mongo-java-driver/pull/1037/files#diff-26a114bb575c4831fa43704d692f44d6fcf087823101af8ab71722d5d9e32276L121